microprofile.jdt/pom.xml

```diff
@@ -36,7 +36,7 @@
     <repository>
       <id>jdt.ls.p2</id>
       <layout>p2</layout>
-      <url>https://download.eclipse.org/jdtls/milestones/1.15.0/repository/</url>
+      <url>https://download.eclipse.org/jdtls/snapshots/repository/latest/</url>
```

Comment:
I think that we had an issue pointing to the snapshots repository in the past, since if we don't update the jdt.ls version then the build breaks when jdt.ls bumps to a new minor version. Do you remember something like this?

Reply:
Maybe with https://repo.eclipse.org/content/repositories/jdtls-snapshots/org/eclipse/jdt/ls/org.eclipse.jdt.ls.tp/ ? Looks like everything prior to 1.16.0 is gone so if we referred to an old enough snapshot, and didn't update for a while, we might run into this ? I don't think we'd have an issue if JDT-LS released though. https://download.eclipse.org/jdtls/snapshots/repository/ I cleaned manually a while ago. and we still have a lot of the older snapshots there.

I know that for https://github.com/eclipse/lsp4mp/blob/master/microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target#L5 , the issue we had is that snapshot artifacts are referred to by URL there, and they can sometimes be cleaned out so it was best to refer to a released one.

Reply:
Okay. I think I got this mixed up with https://github.com/eclipse/lsp4mp/blob/master/microprofile.jdt/org.eclipse.lsp4mp.jdt.tp/org.eclipse.lsp4mp.jdt.target#L5

Reply:
Ok, it just hit me what else could happen. The JDT-LS target file refers to say https://download.eclipse.org/eclipse/updates/4.26-I-builds , which is valid until 4.26 is released (eg. today). After that there is no requirement to retain it (4.25-I-builds are gone). In the past 4.27-I-builds might not even exist yet, but currently it does, and we updated to it. If JDT-LS releases a version that refers to the I-builds of an older release, then once Eclipse releases that I-build, we basically have to jump to a newer JDT-LS.

The solution here is to just stay up to date though.